### ISSUE TITLE
Deploy pyautogui on Heroku 

### ISSUE THREAD

**Issue by AsmitaHari** [asweigart/pyautogui#632]
Hi,

pyautogui works great on my windows local machine. I am trying to deploy to Heroku. The application I am trying to implement is for users with disabilities.  I'm getting the following error Xlib. error.DisplayConnectionError: Can't connect to display ":0":
Is there a way to fix it? Please help.

Thank you!!

**Comment by Miltonbhowmick** [comment]
> 
> 
> PyAutoGUI does not work on systems that are headless. Heroku servers are headless.

Then what will the alternative way to take screenshot? I was deploying my project in pythonanywhere which is also saying Display:0 error.

**Comment by winstxnhdw** [comment]
AFAIK, there is no way to take screenshots on headless systems. Maybe if you Googled harder than I did, you may find a hacky solution.

**Comment by Miltonbhowmick** [comment]
> 
> 
> AFAIK, there is no way to take screenshots on headless systems. Maybe if you Googled harder than I did, you may find a hacky solution.

I am trying it from one day long. Can you give any hints on that solution if you know?

**Comment by winstxnhdw** [comment]
I don't. All the best.

**Comment by gustavofelicidade** [comment]
Hello [Miltonbhowmick](https://github.com/Miltonbhowmick) Have you got a solution? I need to do a similar job with uses pyautogui on heroku and I need that solution. If you still want that solution let me know asap, thanks in advance!

**Comment by Miltonbhowmick** [comment]
> Hello [Miltonbhowmick](https://github.com/Miltonbhowmick) Have you got a solution? I need to do a similar job that uses pyautogui on heroku and I need that solution. If you still want that solution let me know asap, thanks in advance!
> 
> I am thiking in compile the python script.

No, i did not get solution. Actually, you can not use pyautogui in heroku because heroku is a headless server. And pyautogui is not supported of headless server.

**Comment by SamiTesla** [comment]
Yeah I did some research there really isn't an option to get heroku to accept pyautogui maybe you can distribute flash drives?

**Comment by gustavofelicidade** [comment]
I'm thinking use watchdog to verify changes in a folder, kind heroku can
host download  file and then the bot starts when this file get in local sys

SamiTesla ***@***.***> escreveu no dia domingo, 21/08/2022
à(s) 16:44:
